### PR TITLE
feat(i2b2): front end

### PR DESCRIPTION
**Pull request by yvesjaggi**
Only front end for the time being. More to come

### CODE REVIEW

File: charts/i2b2/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#
+# Library charts provide useful utilities or functions for the chart developer. They're included as
+# a dependency of application charts to inject those utilities and functions into the rendering
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
```
Comment (greut) [inline charts/i2b2/Chart.yaml:18]:
```suggestion
version: 0.0.1
```

to match https://github.com/CHORUS-TRE/environments/pull/120

File: charts/i2b2/values.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+image:
+  repository: i2b2/i2b2-web
+  pullPolicy: IfNotPresent
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: "v1.8.0.0002"
```
Comment (greut) [inline charts/i2b2/values.yaml:11]: Why not putting this in the `appVersion` then?

File: charts/i2b2/values.yaml [hunk 3]
```diff
@@ -0,0 +1,89 @@
+# Default values for i2b2.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+image:
+  repository: i2b2/i2b2-web
```
Comment (greut) [inline charts/i2b2/values.yaml:8]: their base image is:

https://hub.docker.com/layers/library/centos/centos7.9.2009/images/sha256-dead07b4d8ed7e29e98de0f4504d87e8880d4347859d839686a31da35a3b532f?context=explore

Screenshot 2024-09-13 at 16-06-28 Image Layer Details - centos centos7 9 2009

😨

File: charts/i2b2/values.yaml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  pullPolicy: IfNotPresent
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: "v1.8.0.0002"
+
+env:
+  url: "https://i2b2.chorus-tre.local"
+  router_base: "/"
+
+imagePullSecrets:
+  - name: regcred
```
Comment (greut) [inline charts/i2b2/values.yaml:18]: it's probably not needed as the image comes from the DockerHub.

File: charts/i2b2/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "1.16.0"
```
Comment (greut) [inline charts/i2b2/Chart.yaml:24]: You may add yourself as the maintainer, see the other charts.

File: charts/i2b2/values.yaml [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  # requests:
+  #   cpu: 100m
+  #   memory: 128Mi
+
+autoscaling:
+  enabled: false
+  minReplicas: 1
+  maxReplicas: 100
+  targetCPUUtilizationPercentage: 80
+  # targetMemoryUtilizationPercentage: 80
```
Comment (greut) [inline charts/i2b2/values.yaml:83]: IMHO, you may drop the whole `autoscaling` block and the `hpa.yaml` file.

**Comment by greut** [comment]
I'm assuming it's safe to put this on the Internet.

File: charts/i2b2/values.yaml (hunk 2 above)
Comment (yvesjaggi) [reply charts/i2b2/values.yaml]: Done

File: charts/i2b2/values.yaml (hunk 3 above)
Comment (yvesjaggi) [reply charts/i2b2/values.yaml:8]: The only image provided for i2b2

File: charts/i2b2/Chart.yaml (hunk 5 above)
Comment (yvesjaggi) [reply charts/i2b2/Chart.yaml]: Done

File: charts/i2b2/values.yaml (hunk 3 above)
Comment (greut) [reply charts/i2b2/values.yaml:8]: I guess it's on their todo list :) I agree that there is nothing you can do here aside telling them that they are on an EOL'd base image.

File: charts/wildfly/values.yaml [hunk 7]
```diff
@@ -110,7 +110,7 @@ wildfly:
   ## @param wildflyPassword WildFly user password
   ## Defaults to a random 10-character alphanumeric string if not set
   ##
-  wildflyPassword: "test"
+  wildflyPassword: "mestest1234"
```
Comment (greut) [inline charts/wildfly/values.yaml:113]: Our CI setup doesn't allow to modify more than one chart at the time.

```bash
git checkout origin/master charts/wildfly/values.yaml
```

Comment (iDmple) [reply charts/wildfly/values.yaml:113]: Yes exactly. @yvesjaggi could you please make 2 PRs, one for i2b2, one for wildfly?

File: charts/i2b2/values.yaml (hunk 3 above)
Comment (iDmple) [reply charts/i2b2/values.yaml:8]: 😨😨😨

**Review by iDmple — CHANGES_REQUESTED** [review]

File: charts/wildfly/values.yaml (hunk 7 above)
Comment (yvesjaggi) [reply charts/wildfly/values.yaml]: outdated

File: charts/i2b2/values.yaml (hunk 4 above)
Comment (iDmple) [reply charts/i2b2/values.yaml:18]: It will be needed as soon as @greut's image is used

**Review by greut — APPROVED** [review]

**Comment by iDmple** [comment]
Closing as we're making a new chart.

**Pull request closed without merging**